http3/server.go

```diff
@@ -436,6 +447,33 @@ func (s *Server) handleConn(conn quic.Connection) error {
 	}).Append(b)
 	str.Write(b)
 
+	var (
+		once   sync.Once
+		mutex  sync.Mutex
+		closed bool
+		// client opened bidirectional stream ID starts at 0 and increase by 4
+		lastID = quic.StreamID(-4)
+	)
+	s.mutex.Lock()
+	s.connections[&conn] = func() {
+		once.Do(func() {
+			mutex.Lock()
+			closed = true
+			b = (&goawayFrame{
+				ID: lastID + 4,
+			}).Append(b[:0])
+			str.Write(b)
+			mutex.Unlock()
```

Comment:
`str.Write` might block indefinitely. You shouldn't hold the mutex that long.

Reply:
The map is there to make sure all connection is closed before `CloseGracefully` returns. It's kept in a map because this way server can begin to send goaway frame this way instead of relying on `AcceptStream` to accept a stream and then write a goaway.

The alternative is to create a goroutine that listens for when the server is closed. But that's even heavier weight than the current method of tracking with map. I guess that's why stdlib uses maps to do the same.